pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherMultiple failed

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherMultiple [failed](https://mesolite.cluster.engflow.com/invocations/default/1cc89ed8-7f52-4b84-8bf2-8421e64809da?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NwYW5jb25maWdjY2wvc3BhbmNvbmZpZ3NxbHdhdGNoZXJjY2w6c3BhbmNvbmZpZ3NxbHdhdGNoZXJjY2xfdGVzdA==) on master @ [7396e5696861e1826e6cfc6455e1c7b3d4806941](https://github.com/cockroachdb/cockroach/commits/7396e5696861e1826e6cfc6455e1c7b3d4806941):


```
W251211 12:32:00.325353 18333 sql/stats/automatic_stats.go:1087  [T10,Vtest-tenant,n1] 2254  failed to create statistics on table 67: create-stats: failed to read query result: query execution canceled
W251211 12:32:00.325859 15836 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T10,Vtest-tenant,rangefeed=sql-watcher-zones-rangefeed,n1,mux_n=1,gen=2] 2255  error closing mux rangefeed client: context canceled
W251211 12:32:00.326525 14488 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T10,Vtest-tenant,n1,rangefeed=sql_instances,mux_n=1,gen=2] 2256  error closing mux rangefeed client: context canceled
W251211 12:32:00.327384 15841 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T10,Vtest-tenant,rangefeed=sql-watcher-descriptor-rangefeed,n1,mux_n=1,gen=2] 2257  error closing mux rangefeed client: context canceled
W251211 12:32:00.327942 15827 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T10,Vtest-tenant,rangefeed=sql-watcher-descriptor-rangefeed,n1,mux_n=1,gen=2] 2258  error closing mux rangefeed client: context canceled
I251211 12:32:00.328472 15649 4@util/log/event_log.go:90  [T10,Vtest-tenant,n1,client=127.0.0.1:36596,hostssl,user=root] 2259 ={"Timestamp":1765456320328463463,"EventType":"client_session_end","InstanceID":1,"Network":"tcp","RemoteAddress":"127.0.0.1:36596","SessionID":"188028f6921c781f0000000000000001","Duration":226920813390}
E251211 12:32:00.331124 44873 jobs/registry.go:920  [T1,Vsystem,n1] 2260  error getting live session: could not update session 010180437f68ea69f24910b6666f813239c911: aborted in DistSender: result is ambiguous: context canceled
    sqlwatcher_test.go:341: 
        	Error Trace:	pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go:341
        	            				src/runtime/asm_amd64.s:1693
        	Error:      	Should be true
        	Test:       	TestSQLWatcherMultiple
    sqlwatcher_test.go:341: 
        	Error Trace:	pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go:341
        	            				src/runtime/asm_amd64.s:1693
        	Error:      	Should be true
        	Test:       	TestSQLWatcherMultiple
W251211 12:32:00.348001 44949 13@kv/kvserver/queue.go:1478  [T1,Vsystem,n1,raftlog,s1,r57/1:/Table/5{4-5}] 2261  queue processing resulted in non-benign error: result is ambiguous: server shutdown
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262  overriding non-cancelation emitted after context cancellation: query execution canceled
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +(1)
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +Wraps: (2) candidate pg code: 57014
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +Wraps: (3) attached stack trace
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  -- stack trace:
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | github.com/cockroachdb/cockroach/pkg/util/cancelchecker.init
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | 	pkg/util/cancelchecker/cancel_checker.go:75
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | runtime.doInit1
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | 	GOROOT/src/runtime/proc.go:7947
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | runtime.doInit
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | 	GOROOT/src/runtime/proc.go:7914
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | runtime.main
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | 	GOROOT/src/runtime/proc.go:256
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | runtime.goexit
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +  | 	src/runtime/asm_amd64.s:1693
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +Wraps: (4) query execution canceled
W251211 12:32:00.215289 44941 sql/execinfra/processorsbase.go:560  [T1,Vsystem,n1,intExec=cancel/pause-requested] 2262 +Error types: (1) *colexecerror.notInternalError (2) *pgerror.withCandidateCode (3) *withstack.withStack (4) *errutil.leafError
E251211 12:32:00.380663 11896 jobs/registry.go:972  [T1,Vsystem,n1] 2263  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: failed to read query result: query execution canceled
E251211 12:32:00.417285 14463 jobs/registry.go:964  [T10,Vtest-tenant,n1] 2264  error expiring job sessions: expire-sessions: failed to read query result: query execution canceled
E251211 12:32:00.418094 14463 jobs/registry.go:972  [T10,Vtest-tenant,n1] 2265  failed to serve pause and cancel requests: txn exec: context canceled
I251211 12:32:00.738147 13959 1@util/log/event_log.go:90  [n1,tenant-orchestration,tenant=test-tenant] 2266 ={"Timestamp":1765456320738138965,"EventType":"tenant_shared_service_stop","NodeID":1,"TenantID":10,"InstanceID":1,"TenantName":"test-tenant"}
W251211 12:32:00.738975 13959 sql/event_log.go:618  [n1,tenant-orchestration,tenant=test-tenant] 2267  failed to start task to save 1 events in eventlog: node unavailable; try another peer
W251211 12:32:00.739612 13959 server/server_sql.go:1809  [T10,Vtest-tenant,n1] 2268  server shutdown without a prior graceful drain
I251211 12:32:00.741380 13959 server/server_controller_channel_orchestrator.go:487  [n1,tenant-orchestration,tenant=test-tenant] 2269  propagate-ungraceful-stop task terminating
I251211 12:32:00.757847 13957 server/server_controller_channel_orchestrator.go:288  [T1,Vsystem,n1] 2270  propagate-close task terminating
W251211 12:32:03.978976 11442 ts/db.go:213  [T1,Vsystem,n1,ts-poll] 2271  error writing time series data: result is ambiguous: server shutdown
W251211 12:32:03.989971 10932 server/server_sql.go:1809  [T1,Vsystem,n1] 2272  server shutdown without a prior graceful drain
W251211 12:32:04.001634 10932 15@kv/kvserver/spanlatch/manager.go:678  [-] 2273  Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-max/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p99.999/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p99.99/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p99.9/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p99/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p90/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p75/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-p50/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-avg/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-count/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.kv.locking-normal-pri-sum/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.elastic_cpu.max_available_nanos/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-max/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p99.999/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p99.99/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p99.9/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p99/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p90/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p75/10s/2025-12-11T12:00:00Z/1], Merge [/System/tsd/cr.node.admission.wait_durations.elastic-cpu-p50/10s/2025-12-11T12:00:00Z/1],... 5444 skipped ..., Merge [/System/tsd/cr.store.totalbytes/10s/2025-12-11T12:00:00Z/1-10], Merge [/System/tsd/cr.store.abortspanbytes/10s/2025-12-11T12:00:00Z/1-10], Merge [/System/tsd/cr.store.keybytes/10s/2025-12-11T12:00:00Z/1-10], Merge [/System/tsd/cr.store.rangekeycount/10s/2025-12-11T12:00:00Z/1-10], Merge [/System/tsd/cr.store.intentage/10s/2025-12-11T12:00:00Z/1-10] has held latch for 3.6s. Some possible causes are slow disk reads, slow raft replication, and expensive request processing.
W251211 12:32:04.005850 10932 15@kv/kvserver/spanlatch/manager.go:678  [-] 2274  EndTxn(parallel commit) [/Table/54/1/1131760580100161537/"legacy_progress"], [txn: 0c5ab7e9] has held latch for 3.8s. Some possible causes are slow disk reads, slow raft replication, and expensive request processing.
I251211 12:32:04.006900 10932 server/server_controller.go:416  [T1,Vsystem,n1] 2275  server controller shutting down
I251211 12:32:04.007232 10932 server/server_controller.go:425  [T1,Vsystem,n1] 2276  waiting for tenant servers to report stopped
--- FAIL: TestSQLWatcherMultiple (247.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherMultiple.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57760

--- a/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/BUILD.bazel
+++ b/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/BUILD.bazel
@@ -26,6 +26,7 @@ go_test(
         "//pkg/sql/catalog/descpb",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/hlc",

--- a/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go
+++ b/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go
@@ -26,6 +26,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -282,6 +283,7 @@ func TestSQLWatcherReactsToUpdates(t *testing.T) {
 // processes, both sequentially and concurrently.
 func TestSQLWatcherMultiple(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderRace(t, "slow test")
 
 	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{

